src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { TodosFilter } from '../../types/TodosFilter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  filter: TodosFilter;
+  onQueryChange: (value: string) => void;
+  onFilterChange: (value: TodosFilter) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  filter,
+  onQueryChange,
+  onFilterChange,
+}) => {
+  const handleReset = () => {
+    onQueryChange('');
+    onFilterChange(TodosFilter.All);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            className="is-capitalized"
+            value={filter}
+            onChange={event => (
+              onFilterChange(event?.target.value as TodosFilter)
+            )}
```

Comment:
Pass a handler as prop from app, which will change ur filterParam to event.target.value and set proper types, it will shorten ur code and will make it look simpler